examples/mlee.py

```diff
@@ -0,0 +1,573 @@
+            ),
+        ]
+
+    def _generate_examples(self, data_files: Path):
+        """
+        Step 6: Create a generator that yields (key, example) of the dataset of interest.
+
+        The arguments to this function come from `gen_kwargs` returned in `_split_generators()`
+
+        The goal of this function is to perform operations on any of the keys of `gen_kwargs` that allow you to extract and process the data.
+
+        The following skeleton does the following:
+
+        - "extracts" abstracts
+        - "extracts" entities, assuming the output is of the form specified in `_info`
+        - "extracts" relations, assuming similarly the output in the form specified in `_info`.
+
+        An assumption in this pseudo code is that the abstract, entity, and relation file all have linking keys.
+        """
+        if self.config.name == _DATASETNAME:
+            txt_files = list(data_files.glob("*txt"))
+            for guid, txt_file in enumerate(txt_files):
+                yield guid, self.parse_brat_file(txt_file)
+        elif self.config.name == _UNIFIED_VIEW_NAME:
+            txt_files = list(data_files.glob("*txt"))
+            for guid, txt_file in enumerate(txt_files):
+                yield guid, self.brat_parse_to_unified_schema(
+                    self.parse_brat_file(txt_file)
+                )
+
+    def brat_parse_to_unified_schema(self, brat_parse: Dict) -> Dict:
+        unified_example = {}
+
+        # identical
+        unified_example["article_id"] = brat_parse["article_id"]
+        unified_example["text"] = brat_parse["text"]
+        unified_example["events"] = brat_parse["events"]
+
+        # get normalizations
+        ref_id_to_normalizations = defaultdict(list)
+        for normalization in brat_parse["normalizations"]:
+            ref_id_to_normalizations[normalization["ref_id"]].append(
+                {
+                    "resource_name": normalization["resource_name"],
+                    "cuid": normalization["cuid"],
+                }
+            )
+
+        # separate entities and event triggers
+        unified_example["entities"] = []
+        unified_example["event_triggers"] = []
+        for ann in brat_parse["text_bound_annotations"]:
+            if ann["type"] in self._ENTITY_TYPES:
+                entity_ann = ann.copy()
+                entity_ann["db_refs"] = ref_id_to_normalizations[ann["id"]]
+                unified_example["entities"].append(entity_ann)
+            else:
+                unified_example["event_triggers"].append(ann.copy())
+
+        # massage relations
+        unified_example["relations"] = []
+        for ann in brat_parse["relations"]:
+            unified_example["relations"].append(
+                {
+                    "head": ann["head"]["ref_id"],
+                    "tail": ann["tail"]["ref_id"],
+                }
+            )
+
+        return unified_example
+
+
+    def parse_brat_file(self, txt_file):
```

Comment:
We might want to move this out to a separate helper module since it looks pretty intimidating for an example file! We could also consider using a standalone [BRAT Standoff file module](https://github.com/snorkel-team/snorkel-extraction/blob/ce9551bd47f1f0a8e52fa88c155e3bb43f636b8b/snorkel/contrib/brat/brat.py#L495).